src/global/useConfig.tsx

```diff
@@ -0,0 +1,18 @@
+import { createContext, useContext, useState } from 'react';
+import { useQuery } from 'react-query';
+
+const ConfigContext = createContext({});
+
+export const ConfigProvider = ({ children }: { children: any }) => {
+	// only need to do this once to get values from runtime api route
+	const [config, setConfig] = useState({});
+	useQuery('config', async () => {
+		const res = await fetch('http://localhost:3000/api');
```

Comment:
for sure - good call! I'm fairly certain the relative '/api' actually works seen as it's the same server instance client pages are rendered from.

Reply:
I am not sure the server-side will be able to tell what the browser route/path will be, so that may not work.

however, surely we can just pass that as an env value, once we figure out that part of this thing, right?

oh wait, chicken and egg!! lol